force-app/main/selector/DonationHistorySelector.cls

```diff
@@ -33,25 +33,31 @@
 * @description Donation History Selector class.
 */
 public inherited sharing class DonationHistorySelector {
-    /** 
-    * @author Salesforce.org
-    * @date 2021
-    * @description method to return OpportunityContactRole data.
-    */
-    public List<Opportunity> getWonOpportunitiesByContactId(Id contactId) {
-        return [
-            SELECT Id, 
-                Name, 
-                CloseDate, 
-                Amount, 
-                Primary_Contact__r.Email, 
-                Primary_Contact__r.Name
-            FROM Opportunity
-            WHERE IsWon = true 
-            AND Primary_Contact__r.Id =:contactId 
-            WITH SECURITY_ENFORCED
-            ORDER BY CloseDate 
-            DESC LIMIT 50
-        ];
-    }
-}
+  /** 
+  * @author Salesforce.org
+  * @date 2021
+  * @description method to return Opportunity and Payment data.
+  */
+  public List<Opportunity> getWonOpportunitiesByContactId(Id contactId) {
+      return [
+          SELECT Id, 
+              Name, 
+              CloseDate, 
+              Amount, 
+              Primary_Contact__r.Email, 
+              Primary_Contact__r.Name,
+                  (SELECT 
+                          npe01__Payment_Method__c 
+                          FROM npe01__OppPayment__r 
```

Comment:
Since we're querying new data from a new object, we should check the user's Read access to `npe01__OppPayment__r` in `DonationHistoryService#hasAccessToDonationHistory()`.